fe/fe-core/src/main/java/com/starrocks/analysis/CreateRoutineLoadStmt.java

```diff
@@ -306,9 +302,10 @@ public static RoutineLoadDesc getLoadDesc(OriginStatement origStmt, Map<String,
         }
 
         // parse the origin stmt to get routine load desc
-        SqlParser parser = new SqlParser(new SqlScanner(new StringReader(origStmt.originStmt), sqlMode));
         try {
-            StatementBase stmt = SqlParserUtils.getStmt(parser, origStmt.idx);
+            List <StatementBase> stmts = com.starrocks.sql.parser.SqlParser.parse(
```

Comment:
If the original stmt stored in image is parsed by old parser. The index will be 0 even though there is comment before the sql. In this case, using the new parser will get the comment line.

Reply:
Hangyuan has fix this by https://github.com/StarRocks/starrocks/pull/7508